File: frontend/src/types/threshold.ts
// Matches ThresholdOut / ThresholdUpdate in openapi.json.

export interface ThresholdOut {
  id: string;
  sector: string;
  metric_key: string;
  label: string;
  warning_value?: number | null;
  critical_value?: number | null;
}

export interface ThresholdUpdate {
  warning_value?: number | null;
  critical_value?: number | null;
}

File: backend/backend/app/schemas/threshold.py
"""Threshold schemas — frozen by contract (ThresholdOut, ThresholdUpdate)."""

from __future__ import annotations

import uuid

from pydantic import BaseModel


class ThresholdOut(BaseModel):
    id: uuid.UUID
    sector: str
    metric_key: str
    label: str
    warning_value: float | None = None
    critical_value: float | None = None


class ThresholdUpdate(BaseModel):
    warning_value: float | None = None
    critical_value: float | None = None
